src/dune_rules/module_compilation.ml

```diff
@@ -218,6 +219,50 @@ let build_cm cctx ~precompiled_cmi ~cm_kind (m : Module.t)
     >>| Action.Full.add_sandbox sandbox))
   |> Memo.Option.iter ~f:Fun.id
 
+let build_melange_js ~cctx m =
```

Comment:
I tried to move this function to `Melange` module but it leads to cyclic dependencies: `Lib_mode->Melange->Compilation_context->Lib_mode`.